spec/Cabin/Seat.vspec

```diff
@@ -182,147 +213,300 @@ Airbag.IsDeployed:
+  type: actuator
+  description: Is switch for less side bolster support engaged.
 
 Switch.Seating:
   type: branch
   description: Describes switches related to the seating of the seat.
+  deprecation: v5.0 - nested properties moved to Seat.Seating.<Property>.
 
 Switch.Seating.IsForwardEngaged:
   datatype: boolean
   type: actuator
   description: Is switch to increase seating length engaged (SingleSeat.Seating.Length).
+  deprecation: v5.0 - refactored to Seat.Seating.IsForwardSwitchEngaged
+
+Seating.IsForwardSwitchEngaged:
+  datatype: boolean
+  type: actuator
+  description: Is switch to increase seating length engaged.
 
 Switch.Seating.IsBackwardEngaged:
   datatype: boolean
   type: actuator
   description: Is switch to decrease seating length engaged (SingleSeat.Seating.Length).
+  deprecation: v5.0 - refactored to Seat.Seating.IsBackwardSwitchEngaged
+
+Switch.IsBackwardSwitchEngaged:
+  datatype: boolean
+  type: actuator
+  description: Is switch to decrease seating length engaged.
 
 Switch.Headrest:
   type: branch
   description: Switches for SingleSeat.Headrest.
+  deprecation: v5.0 - nested properties moved to to Seat.Headrest.<Property>
 
 Switch.Headrest.IsUpEngaged:
   datatype: boolean
   type: actuator
   description: Head rest up switch engaged (SingleSeat.Headrest.Height).
+  deprecation: v5.0 - refactored to to Seat.Headrest.IsUpSwitchEngaged
+
+Headrest.IsUpSwitchEngaged:
+  datatype: boolean
+  type: actuator
+  description: Head rest up switch engaged .
 
 Switch.Headrest.IsDownEngaged:
   datatype: boolean
   type: actuator
   description: Head rest down switch engaged (SingleSeat.Headrest.Height).
+  deprecation: v5.0 - refactored to to Seat.Headrest.IsDownSwitchEngaged
+
+Headrest.IsDownSwitchEngaged:
+  datatype: boolean
+  type: actuator
+  description: Head rest down switch engaged.
 
 Switch.Headrest.IsForwardEngaged:
   datatype: boolean
   type: actuator
   description: Head rest forward switch engaged (SingleSeat.Headrest.Angle).
+  deprecation: v5.0 - refactored to to Seat.Headrest.IsForwardSwitchEngaged
+
+Headrest.IsForwardSwitchEngaged:
+  datatype: boolean
+  type: actuator
+  description: Head rest forward switch engaged.
 
 Switch.Headrest.IsBackwardEngaged:
   datatype: boolean
   type: actuator
   description: Head rest backward switch engaged (SingleSeat.Headrest.Angle).
+  deprecation: v5.0 - refactored to to Seat.Headrest.IsBackwardSwitchEngaged
+
+Headrest.IsBackwardSwitchEngaged:
+  datatype: boolean
+  type: actuator
+  description: Head rest backward switch engaged.  
 
 Switch.Massage:
   type: branch
   description: Switches for SingleSeat.Massage.
+  deprecation: v5.0 - nested properties moved to Seat.<Property>
 
 Switch.Massage.IsIncreaseEngaged:
   datatype: boolean
   type: actuator
   description: Increase massage level switch engaged (SingleSeat.Massage).
+  deprecation: v5.0 - refactored to Seat.IsIncreaseMassageSwitchEngaged
+
+Seat.IsIncreaseMassageSwitchEngaged:
```

Comment:
Remove `Seat.` here and on next I assume

Reply:
True. I suggest that if we introduce sub branches, they must be as explicit as possible. For example, the branch name `SeatMassage` would be preferred over `Massage`. Why? Because such explicit names still hold a meaningful label if they are seen alone. I suggest that the maintenance of the model from now on should focus on two lists:
* List of Features of Interest (FoI).
* List of Properties (or other FoIs) that belong to that FoI.

So, the modeling is simplified as
```mermaid 

graph LR

    FoI_A --> Property_1
    FoI_A --> Property_2
    FoI_A --> Property_N
    FoI_A --> FoI_B
    
    classDef c1 fill:#9370DB,stroke:#333;
    
    class FoI_A,FoI_B c1;
```

The hierarchy of nested branches is still there. The only difference is that, in this way, the concepts make sense and can exist independently from where you want to place it in the tree.

Reply:
I checked the spec again. We actually have a `Massage` property directly linked to `Seat`:
```yaml
Massage:
  datatype: uint8
  type: actuator
  min: 0
  max: 100
  unit: percent
  description: Seat massage level. 0 = off. 100 = max massage.
```
 
The challenge I see is that we have multiple classifications, whereas the tree has one dimension only.

- Seat instances: Vehicle can have multiple instances of `Seat`.
- Seat parts: Each `Seat` has `Headrest`, `Backrest`, and `Seating`.
- Switches in each part
- Massage function (possibly also multiple, one for each seat part).

It seems that `Massage` is modelled right now as if the function would always be centrally controlled. Here are two options I see:

1. Do not include a sub branch `Massage` and attach massage properties directly to the `Seat` branch.
```mermaid
graph LR
    Seat --> MassageProperty_A
    Seat --> MassageProperty_B
    Seat --> MassageProperty_N
    Seat --> Headrest
    Seat --> Backrest
    Seat --> Seating

    classDef c1 fill:#9370DB,stroke:#333;
    
    class Seat,Headrest,Backrest,Seating c1;
```
2. Add a massage sub branch to each seat part as `HeadrestMassage`, `BackrestMassage`, `SeatingMassage`. This option will also imply to deprecate the existing `Seat.Massage` and include that property to each seat part as `HeadrestMassage.MassageIntensity`, `BackrestMassage.MassageIntensity`, `SeatingMassage.MassageIntensity`.
```mermaid
graph LR
    Seat --> Headrest
    Seat --> Backrest
    Seat --> Seating

    Headrest --> HeadrestMassageProperty_A
    Headrest --> HeadrestMassageProperty_B
    Headrest --> HeadrestMassageProperty_N

    Backrest --> BackrestMassageProperty_A
    Backrest --> BackrestMassageProperty_B
    Backrest --> BackrestMassageProperty_N

    Seating --> SeatingMassageProperty_A
    Seating --> SeatingMassageProperty_B
    Seating --> SeatingMassageProperty_N

    classDef c1 fill:#9370DB,stroke:#333;
    
    class Seat,Headrest,Backrest,Seating c1;
```

Which option does sound better @erikbosch ?

Reply:
For now I think keeping it simple makes sense. I.e. not divide it into different components. Maybe you could check with other COVESA representatives from your organization - I think they discussed Massage as part of the capability discussion, that it is an area that may vary between OEMs and maybe not as simple to represent as different intensity values for different sub-components. @paulboyes  - maybe you can summarize something from the capability discussions in this area.

Reply:
So, for now, I will go for option 1.

Reply:
Regarding Massage in the Capabilities discussion... Massage is considered a competitive feature so only the minimum  discussed as a group:

Massage Feature Available - 

- yes/no

Massage Status - 

- on/off

Seat Massage Type

- Off
- Roll
- Knead
- Anti Fatigue
- Pulse
- Stretch
- Lumbar
- Shoulder
- Wave
- Relief
- Glide

When everyone is back in January, we should go through the capabilities and align with VSS